### ISSUE TITLE
🐛 Bug Report: EntityOwnerPicker should only list Users and/or Groups that are actual Owners

### ISSUE THREAD

**Issue by awanlin** [backstage/backstage#18001]
### 📜 Description

The `EntityOwnerPicker` should only list Users and/or Groups that are actual Owners.

PR https://github.com/backstage/backstage/pull/17514 (release as part of `1.15.0-next.0`) changed the `EntityOwnerPicker` to make it perform better at the loss of showing only Users and/or Groups that are actual Owners

### 👍 Expected behavior

The `EntityOwnerPicker` should only list Users and/or Groups that are actual Owners.

### 👎 Actual Behavior with Screenshots

The `EntityOwnerPicker` lists all Users and Groups that are in the Catalog.

### 👟 Reproduction steps

1. Open the Catalog
2. Click on the Owner drop down

**Notice:** All Users and Groups that are in the Catalog are in the list regardless of if they are Owners or not

### 📃 Provide the context for the Bug.

Being able to use the Owner filter to review Ownership was really handy - you could quickly spot cases where there was incorrect teams or in our cases Users (we don't allow Users to be Owners). Now it's just one really large list which we find isn't as useful, especially with all the Users in there.

### 🖥️ Your Environment

Not really environment specific more being on `1.15.0-next.0` or newer

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

No, but I'm happy to collaborate on a PR with someone else

**Comment by awanlin** [comment]
@vinzscam I do appreciate the work you did around this but is there anyway we can still offer the original functionality? We were really using that and surprisingly I got a lot of feedback about this change.

**Comment by sennyeya** [comment]
@awanlin You should be able to make a custom filter with the old functionality, the data from the current set of entities is still available. See https://github.com/backstage/backstage/blob/master/plugins/catalog-react/src/hooks/useEntityListProvider.tsx#L81. When we shift fully to a paginated CatalogTable, I expect you'll need a new workaround.

Can you explain a little bit more what your use case was? I think the shift to the new EntityOwnerPicker makes sense for the pagination work, but it might make sense to show what you're trying to do in a separate UI.

**Comment by vinzscam** [comment]
> @vinzscam I do appreciate the work you did around this but is there anyway we can still offer the original functionality? We were really using that and surprisingly I got a lot of feedback about this change.

sorry for causing troubles @awanlin.
The main goal was to decouple the OwnerPicker from the backend entities, which right now are used to compute the owners' list. However the issue makes sense, as we we recognize that there might be adopters who don't ingest at all users or group information.
As far as I can see, the only doable option that allows us to move towards the use of paginated entities in the future, is to use the facets endpoint to fetch the owners of the entities. The existing logic can be kept as optional, for adopters who have a lot of entities in the catalog.

I will try to address the issue asap, otherwise it would be good to rollback the changes introduced in https://github.com/backstage/backstage/pull/17514 before the main release.

**Comment by awanlin** [comment]
I just want to start off by saying I totally understand the reasons behind the change and those are valid: getting better performance and being able to use pagination is important. I just wanted to make sure we kept this in mind as this may impact others when this goes out with `1.15.0`. Just trying to help us avoid the issue we had with `entityRefs` and the scaffolder as we had a few people on Discord frustrated with that change

@sennyeya, so there's two groups that have mentioned this to me: those who are "spelunking" around the Catalog and the other are those like me who are making sure the Catalog data is correct and catching anything off. Both of these groups had the understanding that the list would be of actual Owners and not simply everyone. 

@vinzscam no worries, I think we are in a tough spot and we needed to make a call one way or the other. If you can find a path to still offer this functionality then that would be great. If not I think as long as we clearly communicate why it was changed (changelogs and release notes) it should be fine. I know we are already adapting to this change internally.

Hopefully this issue will get some comments from others in the Community to help decide the path forward a little

**Comment by sennyeya** [comment]
Yup, the scaffolder issue was definitely my bad, completely get where you're coming from.

I think there's a lot of value in the contextual aspect of the old filter, using the facets endpoint would help with that I think and fix the issue where the existing functionality doesn't let you filter for owners that don't exist in the catalog. It would require another set of calls for showing the display name, but I don't think that's the worst.

That being said, I think it may make sense to pull those experiences apart, on one hand having contextual search and filtering is useful, but for power users a "Catalog Metadata" page may be a better experience, especially for _much_ larger catalogs.

**Comment by awanlin** [comment]
@sennyeya sorry, was not trying to single you out with that comment 😢  

I for sure make mistakes, did it with a show stopping bug in the Linguist plugin when it got released for the first time.

**Comment by awanlin** [comment]
Ran into another case where the old behaviour was helpful. I was trying to find an example for someone on how to define their Component. I had a rough idea of the one I was looking for so had things filtered down to a subset and could have used the owner filter to get this down even further as it would have clued me in once I saw the team name. Sadly all I had was a list of everyone and had to take a different approach
